TestFoundation/TestNSLock.swift

```diff
@@ -41,14 +41,11 @@ class TestNSLock: XCTestCase {
         }
 
         let thread = Thread() {
-            lock.lock()
```

Comment:
This appears to revert #1261 presumably accidentally. Can you remove this file from this PR?